sources/packages/backend/apps/api/src/route-controllers/report/_tests_/e2e/report.institutions.controller.exportReport.e2e-spec.ts

```diff
@@ -0,0 +1,284 @@
+import {
+  ApplicationStatus,
+  Institution,
+  InstitutionLocation,
+  ProgramYear,
+} from "@sims/sims-db";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeEducationProgramOffering,
+  createFakeInstitutionLocation,
+  ensureProgramYearExists,
+  getProviderInstanceForModule,
+  saveFakeApplication,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import {
+  BEARER_AUTH_TYPE,
+  InstitutionTokenTypes,
+  authorizeUserTokenForLocation,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+} from "../../../../testHelpers";
+import { parse } from "papaparse";
+import * as request from "supertest";
+import { SystemUsersService } from "@sims/services";
+import { AppInstitutionsModule } from "../../../../app.institutions.module";
+import { FormService } from "../../../../services";
+import { TestingModule } from "@nestjs/testing";
+
+describe("ReportInstitutionsController(e2e)-exportReport", () => {
+  let app: INestApplication;
+  let appModule: TestingModule;
+  let db: E2EDataSources;
+  let collegeF: Institution;
+  let collegeFLocation: InstitutionLocation;
+  let collegeC: Institution;
+  let collegeCLocation: InstitutionLocation;
+  let systemUsersService: SystemUsersService;
+  let formService: FormService;
+  let programYear: ProgramYear;
+  const REPORT_FORM_NAME = "exportfinancialreports";
+  const PROGRAM_YEAR_PREFIX = 2010;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appModule = module;
+    db = createE2EDataSources(dataSource);
+    // Mock the form service to validate the dry-run submission result.
+    formService = await getProviderInstanceForModule(
+      appModule,
+      AppInstitutionsModule,
+      FormService,
+    );
+    // College F.
+    const { institution: collegeF } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeFLocation = createFakeInstitutionLocation({ institution: collegeF });
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+    // College C.
+    const { institution: collegeC } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeCLocation = createFakeInstitutionLocation({ institution: collegeC });
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+    systemUsersService = nestApplication.get(SystemUsersService);
+    // Program Year for the following tests.
+    programYear = await ensureProgramYearExists(db, PROGRAM_YEAR_PREFIX);
+  });
+
+  it("Should generate the offering details report when a report generation request is made with the appropriate parameters i.e. program year and the offering intensity", async () => {
```

Comment:
As a suggestion, it can be adapted as below.
```
Should generate the offering details report when a report generation request is made with the appropriate program year and offering intensity.
```